lib/client.js

```diff
@@ -232,6 +232,28 @@ var Client = function(options) {
   this._context = options.context || null;
   this.ready = false;
 
+  var WHITELISTED_ORIGINS = [
+    /.*zendesk\.com.*/gi,
+    /.*zd-staging\.com.*/gi,
+    /.*zd-dev\.com.*/gi,
+    /.*zd-master\.com.*/gi,
+    /.*zendesk-staging\.com.*/gi,
+    /.*dashboard\.zopim\.com.*/gi,
+    /.*dashboard\.zopim\.org.*/gi
+  ];
+
+  var i;
+
+  for (i = 0; i < WHITELISTED_ORIGINS.length; i++) {
+    if (WHITELISTED_ORIGINS[i].test(this._origin)) {
+      break;
+    }
+  }
```

Comment:
Could we extract this into a function that returns `true` if the origin passed matches any regex, then if the function returns `true` throw the exception below?

Reply:
Could we add a test please?

Reply:
I'll move it to a separate function. I think I also need to add `*.apps.zdusercontent.com` to the list.